eZ/Publish/API/Repository/LocationService.php

```diff
@@ -216,10 +217,11 @@ public function deleteLocation(Location $location);
      * Instantiates a new location create class.
      *
      * @param mixed $parentLocationId the parent under which the new location should be created
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType|null $contentType
```

Comment:
it would be nice to have a comment explaining why this param was added, as it is not super intuitive... (tbh I would have preferred that the sorting order was grabbed and injected inside LocationService::createLocation, which already has access to the $contentInfo...)

Reply:
We could do that, however then we need to change struct to set values to null by default instead of, and document it there how it is supposed to work if not defined:
https://github.com/ezsystems/ezpublish-kernel/blob/7.5/eZ/Publish/API/Repository/Values/Content/LocationCreateStruct.php#L44-L60

@alongosz WDT? Would greatly simplify use/DX.

Reply:
@andrerom indeed the change in default values to LCS members would be a small BC break - but imho one in the good direction. It would allow extreme flexibility (by default use sorting from content you are adding a location to, but can override it if desired) with a clean and simple api

Reply:
> (tbh I would have preferred that the sorting order was grabbed and injected inside LocationService::createLocation, which already has access to the $contentInfo... adding it here does allow to have discordant data)

I agree, seems like better place, responsibility-wise.

Reply:
@barw4 All clear on how this can be done for adapting PR?

Reply:
@andrerom hopefully, adapted above solution in https://github.com/ezsystems/ezpublish-kernel/pull/2977/commits/0789402be0da50ddd24933fd1ee0537851161a05 and https://github.com/ezsystems/ezpublish-kernel/pull/2977/commits/be90ca9c9308cd5df1c120e037ef60476c500fbc

I presume my `admin-ui` PR can be now closed:
https://github.com/ezsystems/ezplatform-admin-ui/pull/1270